packages/block-library/src/tag-cloud/edit.js

```diff
@@ -69,7 +69,7 @@ function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 			disabled: true,
 		};
 		const taxonomyOptions = map(
-			filter( taxonomies, 'show_cloud' ),
+			taxonomies?.filter( ( tax ) => !! tax.show_cloud ),
```

Comment:
Although very edge case-y, `taxonomies` could be `null`:

https://github.com/WordPress/gutenberg/blob/33d84b036592a5bf31af05b7710f3b2b14163dc4/packages/core-data/src/queried-data/selectors.js#L99